datasource/hkobserverfactory.js

```diff
@@ -18,6 +19,26 @@ for (let key in classes)
 	clients[C.getType()] = C;
 }
 
+/**
+ * Instantiate an observer client.
+ * This method will the target HKBase which observer configuration it supports (REST or RabbitMQ)
+ * and will instantiate the appropriate client with the provided configurations.
+ * @param {string} basePath HKBase basePath that is used to communicate with the server
+ * @param {Object} observerOptions Observer options that are used to initialize the observer client
+ * @param {boolean} observerOptions.isObserverService if true, the factory method will assume that it 
```

Comment:
Hmm. Currently the REST service is named observer service, but it has an internal service that is named notification service (which would not show up for the clients). Therefore I believe observer service is appropriate, but I would not bother to change to Notification Service if it makes more sense (although I would also have to change the source code of the service itself to use the same name).

Reply:
Ok, perhaps my question is: isn't more appropriate to call the RESTFull service Notification Service rather than Observer Service? Isn't its main goal to process changes in the HKB in order to notify observers?

Reply:
I guess it would make sense, but I'm not yet convinced that observer service is a bad name for the service. 
One of the reasons this name makes sense to me because it is a service that *observes* the HKBase.

The fact that it redirects the notifications to the clients would make it reasonable to rename it to notification service, but, if I were to rename it to something that conveys the notification functionality more clearly, I would call it "NotificationFilterService" or some other long name.

Nonetheless, HKBaseObserverService feels more intuitive to me, specially because HKLib and HKPy classes are named ObserverClient. Therefore, the ObserverServer can be HKBase itself (when no ObserverConfiguration is provided or the HKBaseObserverService, otherwise).

In other words, the name HKBaseObserverService makes more sense to me because, in my understanding, what the service actually "serves" are observers, that are managed by the service. The delivery of notifications that correspond to each of these observers is only a consequence of that.
In addition, all the service endpoints are related to CRUD operations of observers, and none of them mention notifications, which could be an additional argument in favor of this name.

Does that make any sense?

Reply:
I agree with you, perhaps the issue is with the naming of the REST service's resource. I'm not sure `Observer` is the best name; Notification Filters seems like a more apt descriptor. 

I'd have to be more familiar with the openapi in order to have a position wrt your closing arguments. 

I say we leave it at that for now.